assets/__tests__/rendered_test.js

```diff
@@ -0,0 +1,191 @@
+
+    </span>
+  </div>
+</div>
+      `.trim()
+      )
+    })
+  })
+
+  describe('toOutputBuffer', () => {})
+
+  describe('comprehensionToBuffer', () => {})
+
+  describe('dynamicToBuffer', () => {})
+})
+
+const simpleDiff1 = {
+  '0': 'cooling',
+  '1': 'cooling',
+  '2': '07:15:03 PM',
+  '3': '72',
+  '4':
+    '<div conn id="phx-dafeTrsp/m0=:DemoWeb.WeatherLive"\n     data-phx-parent-id="phx-dafeTrsp/m0="\n     data-phx-view="DemoWeb.WeatherLive"\n     data-phx-session="STUBBED"></div>\n<div class="phx-loader"></div>\n',
+  static: [
+    '<div class="thermostat">\n  <div class="bar ',
+    '">\n    <a href="#" phx-click="toggle-mode">',
+    '</a>\n    <span>',
+    '</span>\n  </div>\n  <div class="controls">\n    <span class="reading">',
+    '</span>\n    <button phx-click="dec" class="minus">-</button>\n    <button phx-click="inc" class="plus">+</button>\n    <span class="weather">\n      ',
+    '\n    </span>\n  </div>\n</div>\n',
+  ],
+}
+
+const simpleDiff2 = {
+  '2': '07:15:04 PM',
+}
+
+const simpleDiffResult = {
+  '0': 'cooling',
+  '1': 'cooling',
+  '2': '07:15:04 PM',
+  '3': '72',
+  '4':
+    '<div conn id="phx-dafeTrsp/m0=:DemoWeb.WeatherLive"\n     data-phx-parent-id="phx-dafeTrsp/m0="\n     data-phx-view="DemoWeb.WeatherLive"\n     data-phx-session="STUBBED"></div>\n<div class="phx-loader"></div>\n',
+  static: [
+    '<div class="thermostat">\n  <div class="bar ',
+    '">\n    <a href="#" phx-click="toggle-mode">',
+    '</a>\n    <span>',
+    '</span>\n  </div>\n  <div class="controls">\n    <span class="reading">',
+    '</span>\n    <button phx-click="dec" class="minus">-</button>\n    <button phx-click="inc" class="plus">+</button>\n    <span class="weather">\n      ',
+    '\n    </span>\n  </div>\n</div>\n',
+  ],
+}
+
+const deepDiff1 = {
+  '0': {
+    '0': {
+      dynamics: [['user1058', '1'], ['user99', '1']],
+      static: ['        <tr>\n          <td>', ' (', ')</td>\n        </tr>\n'],
+    },
+    static: [
+      '  <table>\n    <thead>\n      <tr>\n        <th>Username</th>\n        <th></th>\n      </tr>\n    </thead>\n    <tbody>\n',
+      '    </tbody>\n  </table>\n',
+    ],
+  },
+  '1': {
+    dynamics: [
+      [
+        'asdf_asdf',
+        'asdf@asdf.com',
+        '123-456-7890',
+        '<a href="/users/1">Show</a>',
+        '<a href="/users/1/edit">Edit</a>',
+        '<a href="#" phx-click="delete_user" phx-value="1">Delete</a>',
+      ],
+    ],
+    static: [
+      '    <tr>\n      <td>',
+      '</td>\n      <td>',
+      '</td>\n      <td>',
+      '</td>\n\n      <td>\n',
+      '        ',
+      '\n',
+      '      </td>\n    </tr>\n',
+    ],
+  },
+  '2': '<a href="/users/new">New User</a>',
+  static: [
+    '<h2>Listing Users</h2>\n\n',
+    '\n<table>\n  <thead>\n    <tr>\n      <th>Username</th>\n      <th>Email</th>\n      <th>Phone #</th>\n\n      <th></th>\n    </tr>\n  </thead>\n  <tbody>\n',
+    '  </tbody>\n</table>\n\n<span>',
+    '</span>\n',
+  ],
```

Comment:
Anything we can cut from this object that isn't critical to the test?

Reply:
I don't want to touch this one cause it's so large and the data structure is fragile. Thoughts?